conan/tools/cmake/toolchain/blocks.py

```diff
@@ -919,6 +920,58 @@ def _is_apple_cross_building(self):
         return os_host in ('iOS', 'watchOS', 'tvOS', 'visionOS') or (
                 os_host == 'Macos' and (arch_host != arch_build or os_build != os_host))
 
+    def _get_darwin_version(self, system_name, os_version):
```

Comment:
I think this could be made more compact, something like:

```
    def _get_darwin_version(self, os_name, os_version):

        os_version = Version(os_version).major if os_name != "Macos" or (os_name == "Macos" and Version(
            os_version) >= Version("11")) else os_version

        version_mapping = {
            "iOS": {
                "7": "14", "8": "14", "9": "15",
                ...
            },
            "watchOS": {
                "4": "17", "5": "18", "6": "19",
                ...
            },
            ...
        }
        return version_mapping.get(os_name, {}).get(os_version)
```